packages/block-library/src/comments-pagination/block.json

```diff
@@ -32,6 +32,6 @@
 			}
 		}
 	},
-	"editorStyle": "wp-block-comments-pagination-editor",
-	"style": "wp-block-comments-pagination"
+	"editorStyle": "wp-block-query-pagination-editor",
```

Comment:
Does it mean it reuses the styles from the Query Pagination block?

I'm wondering how much overlap there really is between comments and posts. For comments, I think there are now links for older/newer comments instead of next/previous page. It doesn't mean we can't change it. However, it would be great to clarify the design.

Reply:
This doesn't seem right. If we need some styles we should copy them and add the specificity of the comments block.

Reply:
This reuses the styles due to [this function ](https://github.com/WordPress/gutenberg/blob/7cbba2646a522738fd95894db8240e4d3d5ac1d3/lib/compat/wordpress-5.9/blocks.php#L23), it is applying a class to the arrow with that name.

Should we create a new function to render the arrow for comments?

Could I add a new variable to set if it is a query or a comment instead?

Reply:
I'm still confused why we need a change here. And actually this results in not loading this block styles (change comments-pagination/styles.scss and check the frontend). I think we can reuse the `get_query_pagination_arrow` function which adds a css class like `wp-block-query-pagination-...`,  just copy the related styles from QueryPagination styles with any small possible needed adjustments - the styles will be wrapped under each block's base css class, and finally remove [this change](https://github.com/WordPress/gutenberg/pull/36562/files#diff-2c42b7415c23e59cddb6d85e8474729f9c2b4e3449ffc0571b3b4771e706db87R23):

`get_block_wrapper_attributes( array( 'class' => 'wp-block-query-pagination' ) )`

You made the above change for this styling issues, no? If not can you share the reason for adding it?

Reply:
Thanks for your patience and sorry for messing this issue so much 😓 

The reason of `get_block_wrapper_attributes( array( 'class' => 'wp-block-query-pagination' ) )` is because we are not getting the styles in editor / frontend, due to the block has the parent class `wp-block-**comment**-pagination` instead of `wp-block-**query**-pagination` on the parent.

For the arrow, the function `get_query_pagination_arrow` adds the child class `wp-block-query-pagination-$pagination_type-arrow`.

So, in order to have the correct styling, we need `.wp-block-query-pagination .wp-block-query-pagination-type-arrow`

The SCSS files on the pagination block, being the same as query pagination, won't be loaded if we only have a Comment Query Loop. That's the reason why we need them to be duplicated in both blocks.

Anyway, let's better make a small refactor to have everything clear, I'll keep you updated!

Reply:
No worries at all @c4rl0sbr4v0, there are a lot of subtle things here. You are doing great work here not messing anything! 😄

Reply:
The more I read about those styles, the more I think that we should build a completely independent logic for the Comments Pagination block. The styles used for Query Pagination are tied to the default class names generated for those blocks and it's only 50-ish lines of code:
https://github.com/WordPress/gutenberg/blob/trunk/packages/block-library/src/query-pagination/style.scss

The PHP implementation for the `get_query_pagination_arrow` is 20 lines of code:
https://github.com/WordPress/gutenberg/blob/78c7f0c7a9c3d69979d0633375325104fb03b86d/lib/compat/wordpress-5.9/blocks.php#L23-L43

It hardcodes the class name and the context entry that is tied to the Query Pagination block.